File: app/src/main/java/com/example/ogan/codedenim/LearningPathDetailActivity.java
package com.example.ogan.codedenim;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.ogan.codedenim.courses.CoursesByCategories;
import com.squareup.picasso.Picasso;

public class LearningPathDetailActivity extends AppCompatActivity {

    private String studentType;
    private int categoryId;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_learning_path_detail);

        ImageView image = findViewById(R.id.learningPathPic_img);
        TextView name = findViewById(R.id.learningPathName_tv);
        TextView description = findViewById(R.id.learningPathDescription_tv);
        TextView amount = findViewById(R.id.learningPathPrice_tv);
        Button enrollBtn = findViewById(R.id.learningPath_enroll_btn);

        Intent intent = getIntent();
        String imagePath = intent.getStringExtra("ImageLocation");
        final String learningPathName = intent.getStringExtra("CategoryName");
        String learningPathDesc = intent.getStringExtra("CategoryDescription");
        String learningPathPrice = intent.getStringExtra("CategoryPrice");
        studentType = intent.getStringExtra("StudentType");
        categoryId = intent.getIntExtra("CategoryId", 0);
        setTitle(learningPathName);

        Picasso.with(getApplicationContext()).load(imagePath).into(image);
        name.setText(learningPathName);
        description.setText(getResources().getString(R.string.learning_path_desc, learningPathDesc));
        amount.setText(getResources().getString(R.string.learning_path_price, learningPathPrice));
        if (studentType.equals("Corper")){
            enrollBtn.setText("Enroll");
        }

        enrollBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), CoursesByCategories.class);
                intent.putExtra("CategoryId", categoryId);
                intent.putExtra("CategoryName", learningPathName);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/example/ogan/codedenim/LearningPathDetailActivity.kt
package com.example.ogan.codedenim

import android.net.Uri
import android.os.Bundle
import android.support.customtabs.CustomTabsIntent
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.widget.ImageView
import android.widget.TextView
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_learning_path_detail.*

class LearningPathDetailActivity : AppCompatActivity() {

    private var categoryId: Int = 0
    private var learningPathName: String? = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_learning_path_detail)

        val image = findViewById<ImageView>(R.id.learningPathPic_img)
        val name = findViewById<TextView>(R.id.learningPathName_tv)
        val description = findViewById<TextView>(R.id.learningPathDescription_tv)
        val amount = findViewById<TextView>(R.id.learningPathPrice_tv)

        val intent = intent
        val imagePath = intent.getStringExtra("ImageLocation")
        learningPathName = intent.getStringExtra("CategoryName")
        val learningPathDesc = intent.getStringExtra("CategoryDescription") ?: "No description"
        val learningPathPrice = intent.getStringExtra("CategoryPrice")
        categoryId = intent.getIntExtra("CategoryId", 0)
        title = learningPathName

        Picasso.with(applicationContext).load(imagePath).into(image)
        name.text = learningPathName
        description.text = resources.getString(R.string.learning_path_desc, learningPathDesc)
        amount.text = resources.getString(R.string.learning_path_price, learningPathPrice)
        learningPath_enroll_btn.text =  resources.getString(R.string.learning_path_pay_button, learningPathPrice)

        //val user = UserSessionManager.getUserDetails()
        //val email = user[UserSessionManager.KEY_EMAIL]

        learningPath_enroll_btn.setOnClickListener {
            val url = "http://codedenim.azurewebsites.net/CourseCategories/CategoryDetails/$categoryId"
            val builder = CustomTabsIntent.Builder()
            builder.setToolbarColor(ContextCompat.getColor(this, R.color.colorPrimaryDark))
            val customTabIntent = builder.build()
            customTabIntent.launchUrl(this, Uri.parse(url))
            finish()
        }


    }
}
